cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -877,6 +857,28 @@ func (a *StaticAutoscaler) deleteCreatedNodesWithErrors() bool {
 	return deletedAny
 }
 
+// zeroOrMaxNodesToDelete returns false if the node group is not a "ZeroOrMaxNodeScaling" node group.
+// Returns true for "ZeroOrMaxNodeScaling" node group, together with an override list of nodes to delete.
+// Node deletion for a "ZeroOrMaxNodeScaling" node group is atomic and should delete all nodes.
+//
+// Ideally, this would be handled by the node group itself, but with current setup, it would have to be implemented by every cloud provider.
```

Comment:
Not sure if this line is super helpful in this specific place, maybe you can move it to the definition of ZeroOrMaxScaling option? Or, if you have specific ideas how to improve/refactor it to deprecate this option and move to a more appropriate place, maybe tracking it as an issue would be more appropriate?

Reply:
Yeah, I guess that was me venting mostly, removing. I honestly don't have a good idea how to make this better easily...

Reply:
Yeah, the zero-or-max-node-scaling groups are quite annoying:/ But the zero-to-max is very much part of CA logic, not cloudprovider logic. CA needs to know about it and account for it when it runs scale-up/scale-down simulations, checks limits etc.